wrappers/golang/internal/generator/poseidon/generate.go

```diff
@@ -0,0 +1,34 @@
+package poseidon
+
+import (
+	"path"
+
+	generator "github.com/ingonyama-zk/icicle/v2/wrappers/golang/internal/generator/generator_utils"
+)
+
+var poseidonTemplates = map[string]string{
+	"src":    "poseidon/templates/poseidon.go.tmpl",
+	"test":   "poseidon/templates/poseidon_test.go.tmpl",
+	"header": "poseidon/templates/poseidon.h.tmpl",
+}
+
+func Generate(baseDir, additionalDirPath, field, fieldPrefix string) {
+
+	data := struct {
+		PackageName    string
+		Field          string
+		FieldPrefix    string
+		BaseImportPath string
+	}{
+		"poseidon",
+		field,
+		fieldPrefix,
+		baseDir,
+	}
+
+	testPath := poseidonTemplates["test"]
```

Comment:
Not necessary to create a new var here

Reply:
yeah not sure why it's there. Most likely copy and pasted without paying too much attention, will fix.